packages/commonwealth/server/routes/updateSiteAdmin.ts

```diff
@@ -0,0 +1,58 @@
+import { AppError } from 'common-common/src/errors';
+import type { Response } from 'express';
+import type { DB } from '../models';
+import type { TypedRequestBody } from '../types';
+import { success } from '../types';
+import { Op } from 'sequelize';
+
+enum PromoteUserErrors {
+  NoUser = 'Must supply a user address',
+  NotAdmin = 'You do not have permission to promote users',
+}
+
+type PromoteUserReq = {
+  address: string;
+  siteAdmin: boolean;
+};
+
+const updateSiteAdmin = async (
+  models: DB,
+  req: TypedRequestBody<PromoteUserReq>,
+  res: Response
+) => {
+  const { address, siteAdmin } = req.body;
```

Comment:
Any reason why this route is queried by address rather than providing a user or profile?

Reply:
That way the frontend does not need to do anything other than provide the address as a string. This makes it much easier for the BD team to promote users; they don't have to hunt down the user id or profile id and the frontend doesn't have to fetch that information either. Since all addresses will be attached to the same user id I figured this was acceptable and not a security risk.

Reply:
Fine by me.